yt/frontends/exodus_ii/data_structures.py

```diff
@@ -336,7 +336,30 @@ def _read_connectivity(self):
         connectivity = []
         with self._handle.open_ds() as ds:
             for i in range(self.parameters["num_meshes"]):
-                connectivity.append(ds.variables["connect%d" % (i + 1)][:].astype("i8"))
+                var = ds.variables["connect%d" % (i + 1)][:].astype("i8")
+                try:
+                    elem_type = var.elem_type.lower()
+                    if elem_type == "nsided":
+                        arbitrary_polyhedron = True
+                    elif elem_type == "nfaced":
+                        raise NotImplementedError(
+                            "3D arbitrary polyhedra are not implemented yet"
+                        )
+                    else:
+                        arbitrary_polyhedron = False
+                except AttributeError:
+                    arbitrary_polyhedron = False
+
+                conn = var[:].astype("i8")
```

Comment:
Seems redundant since you're already imposing the dtype to `i8` earlier
```suggestion
                conn = var[:]
```

Reply:
Agreed.